System.Containers.Tasks/CreateNewImage.cs

```diff
@@ -1,105 +1,146 @@
+
+    /// <summary>
+    /// The registry to push to.
+    /// </summary>
+    [Required]
+    public string OutputRegistry { get; set; }
+
+    /// <summary>
+    /// The name of the image to push to the registry.
+    /// </summary>
+    [Required]
+    public string ImageName { get; set; }
+
+    /// <summary>
+    /// The tag to associate with the image.
+    /// </summary>
+    public string ImageTag { get; set; }
+
+    /// <summary>
+    /// The directory for the build outputs to be published.
+    /// Constructed from "$(MSBuildProjectDirectory)\$(PublishDir)"
+    /// </summary>
+    [Required]
+    public string PublishDirectory { get; set; }
+
+    /// <summary>
+    /// The working directory for the container.
+    /// </summary>
+    [Required]
+    public string WorkingDirectory { get; set; }
+
+    /// <summary>
+    /// The entrypoint for the container.
+    /// </summary>
+    [Required]
+    public string Entrypoint { get; set; }
+
+    /// <summary>
+    /// Arguments to pass alongside Entrypoint.
+    /// </summary>
+    public string EntrypointArgs { get; set; }
+
+    public CreateNewImage()
     {
-        /// <summary>
-        /// Base image name.
-        /// </summary>
-        [Required]
-        public string BaseImageName { get; set; }
-
-        [Required]
-        public string BaseImageTag { get; set; }
-
-        [Required]
-        public string InputRegistryURL { get; set; }
-
-        [Required]
-        public string OutputRegistryURL { get; set; }
-
-        /// <summary>
-        /// Constructed from "$(MSBuildProjectDirectory)\$(PublishDir)"
-        /// </summary>
-        [Required]
-        public string PublishDirectory { get; set; }
-
-        /// <summary>
-        /// $(ContainerWorkingDirectory)
-        /// </summary>
-        [Required]
-        public string WorkingDirectory { get; set; }
-
-        [Required]
-        public string NewImageName { get; set; }
-
-        [Required]
-        public string Entrypoint { get; set; }
-
-        /// <summary>
-        /// Arguments to pass alongside Entrypoint.
-        /// </summary>
-        public string EntrypointArgs { get; set; }
-
-
-
-        /// <summary>
-        /// CreateNewImage needs to:
-        /// 1. Pull a base image (needs parameters: URL, BaseImage, BaseImageTag)
-        /// 2. Add output of build as a new layer
-        /// 3. Push image back to some registry (needs parameters: OutputURL, NewName, EntryPoint)
-        /// </summary>
-        /// <returns></returns>
-        public override bool Execute()
+        BaseRegistry = "";
+        BaseImageName = "";
+        BaseImageTag = "";
+        OutputRegistry = "";
+        ImageName = "";
+        ImageTag = "";
+        PublishDirectory = "";
+        WorkingDirectory = "";
+        Entrypoint = "";
+        EntrypointArgs = "";
+    }
+
+
+    public override bool Execute()
+    {
+        if (string.IsNullOrEmpty(PublishDirectory))
```

Comment:
Yeah this is a very conservative check, in case for some reason the publishdir was an empty value. Which I suppose would fail the Directory.Exists check anyway. Will simplify